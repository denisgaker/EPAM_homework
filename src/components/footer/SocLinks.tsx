import React, { FC } from 'react';
import withStyles, { WithStylesProps } from 'react-jss';
import socLinks from './socLinks.constants';
import styles from './style';
import icons from './img/icons';

interface StyledSocLinksProps extends WithStylesProps<typeof styles> {}

const SocLinks:FC<StyledSocLinksProps> = ({ classes }) => (
    <div className={classes.soc}>
      {socLinks.map((sl) => (
        <a href={sl.link}
           target={sl.target}
           title={sl.desc}
           key={sl.key}>
            <img
              src={icons.filter((ic) => ic.name === sl.name)[0].link}
              alt={sl.desc}
            />
        </a>
      ))}
    </div>
);

const StyledSocLink = withStyles(styles)(SocLinks);

export default StyledSocLink;
