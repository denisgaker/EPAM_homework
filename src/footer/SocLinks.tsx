import * as React from 'react';
import Datats from '../Data';

const FooterData = Datats.socialNetworkLinks;

const SocLinks: React.FC = () => (
    <div className="soc">
      {FooterData.map((fd) => (
        <a href={fd.link} target={fd.target} title={fd.desc} className={fd.name} key={fd.key}></a>
      ))}
    </div>
);

export default SocLinks;
