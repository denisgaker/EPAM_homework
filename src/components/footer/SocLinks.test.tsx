import * as React from 'react';
import * as renderer from 'react-test-renderer';
import SocLinks from './SocLinks';

test('Social network links', () => {
  const component = renderer.create(<SocLinks />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
