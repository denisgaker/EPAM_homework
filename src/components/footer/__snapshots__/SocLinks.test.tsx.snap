// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social network links 1`] = `
<div
  className="soc"
>
  <a
    className="vk"
    href="https://vk.com"
    target="_blanck"
    title="Переход на социальную сеть Вконтакте"
  />
  <a
    className="facebook"
    href="https://facebook.com"
    target="_blanck"
    title="Переход на социальную сеть Facebook"
  />
  <a
    className="instagram"
    href="https://instagram.com"
    target="_blanck"
    title="Переход на социальную сеть Instagram"
  />
  <a
    className="youtube"
    href="https://youtube.com"
    target="_blanck"
    title="Переход на видеохостинг Youtube"
  />
</div>
`;
